### ISSUE TITLE
Exclude DevelopmentDependency from bom.xml

### ISSUE THREAD

**Issue by cmenzi** [CycloneDX/cyclonedx-dotnet#44]
Later `.csproj` files supports package references with are just used in development scope.

They usually looks like: 

```xml
<PackageReference Include="Foo" Version="1.0.0">
   <PrivateAssets>all</PrivateAssets>
   <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```

This could be Roslyn.Analyzers, SourceLink packages, ... which will never go to production.

see [DevelopmentDependency](https://github.com/NuGet/Home/wiki/DevelopmentDependency-support-for-PackageReference)

I think it would be good to exclude such dependencies.

**Comment by stevespringett** [comment]
Agreed. The majority of CycloneDX implementations for other languages (npm, maven, etc) allow the user to specify the scope or the ability to explicitly exclude dev dependencies. This would be a good enhancement.

**Comment by coderpatros** [comment]
@cmenzi and @stevespringett I'm about to merge PR #188 "Scope development dependencies as excluded"

My thoughts are that this supersedes this issue. Although it would be easy enough to implement this, I don't think it adds value.

I think this should probably be handled by the tools, like Dependency-Track, that consume SBOMs.

What are your thoughts?

**Comment by stevespringett** [comment]
Dependency-Track currently does not account for scope, but should. Scoping is certainly the correct thing to do. The npm and maven plugins however allow for complete exclusion.

**Comment by coderpatros** [comment]
@cmenzi I've just implemented this in PR #190. It isn't released yet. But it will be in the next release so I'm going to close this issue.
